src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseDataSourceServiceImpl.java

```diff
@@ -316,10 +316,44 @@ public DS findByUuidUnsecured(String uuid) throws NotExistException {
     @Override
     public Page<DS> suggestDataSource(final String query, final Long studyId, final Long userId,
                                       PageRequest pageRequest) {
-
+        List<DataSourceStatus> BAD_STATUSES = Arrays.asList(DataSourceStatus.DELETED, DataSourceStatus.DECLINED);
         final String[] split = query.trim().split(" ");
-        String suggestRequest = "%(" + String.join("|", split) + ")%";
-        return doSuggestDataSource(suggestRequest, userId, studyId, pageRequest);
+
+        CriteriaBuilder cb = this.entityManager.getCriteriaBuilder();
+        CriteriaQuery<DS> cq = (CriteriaQuery<DS>) cb.createQuery(RawDataSource.class);
```

Comment:
Why RawDataSource? It used DataSource previously.

What about generics? - "DS" can be changed, but entity type is hardcoded. Why not create some method like "DS getEmptyDataSource()"?